analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Conversion.cs

```diff
@@ -18,18 +18,40 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using SonarAnalyzer.SymbolicExecution.Constraints;
+
 namespace SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors;
 
-internal sealed class Conversion : SimpleProcessor<IConversionOperationWrapper>
+internal sealed class Conversion : MultiProcessor<IConversionOperationWrapper>
 {
     protected override IConversionOperationWrapper Convert(IOperation operation) =>
         operation.ToConversion();
 
-    protected override ProgramState Process(SymbolicContext context, IConversionOperationWrapper conversion) =>
-        context.State[conversion.Operand] is { } value && IsBuildIn(conversion.OperatorMethod) // Built-in conversions only
-            ? context.SetOperationValue(value)
-            : context.State;
+    protected override ProgramState[] Process(SymbolicContext context, IConversionOperationWrapper conversion) =>
+        IsNotBuildIn(conversion.OperatorMethod) ? context.State.ToArray() : StateForBuildInConversion(context, conversion);
+
+    private static bool IsNotBuildIn(ISymbol symbol) =>
+        symbol?.ContainingType.IsAny(KnownType.PointerTypes) is false;
+
+    private static ProgramState[] StateForBuildInConversion(SymbolicContext context, IConversionOperationWrapper conversion) =>
+        IsNotUncertainTryCast(conversion)
+            ? StateForRegularConversion(context, context.State[conversion.Operand])
+            : StateForUncertainTryCast(context, context.State[conversion.Operand]);
```

Comment:
I wouldn't extract this method... The name is artificial anyway. Just embed it as it was with `if() { } else { }`